rust/connlib/tunnel/src/client.rs

```diff
@@ -1307,7 +1499,68 @@ mod tests {
         })
     }
 
+    fn cidr_resource(addr: &str) -> ResourceDescription {
+        ResourceDescription::Cidr(ResourceDescriptionCidr {
+            id: cidr_id(),
+            address: addr.parse().unwrap(),
+            name: "foo".to_string(),
+        })
+    }
+
+    fn cidr_resource_dns_id(addr: &str) -> ResourceDescription {
+        ResourceDescription::Cidr(ResourceDescriptionCidr {
+            id: dns_id(),
+            address: addr.parse().unwrap(),
+            name: "foo".to_string(),
+        })
+    }
+
+    fn dns_resource(addr: &str) -> ResourceDescription {
+        ResourceDescription::Dns(ResourceDescriptionDns {
+            id: dns_id(),
+            address: addr.to_string(),
+            name: "bar".to_string(),
+        })
+    }
+
+    fn additional_resource(addr: &str) -> ResourceDescription {
+        ResourceDescription::Cidr(ResourceDescriptionCidr {
+            id: additional_id(),
+            address: addr.parse().unwrap(),
+            name: "baz".to_string(),
+        })
+    }
```

Comment:
```suggestion
    fn cidr_foo_resource(addr: &str) -> ResourceDescription {
        ResourceDescription::Cidr(ResourceDescriptionCidr {
            id: cidr_id(),
            address: addr.parse().unwrap(),
            name: "foo".to_string(),
        })
    }

    fn cidr_bar_resource_dns(addr: &str) -> ResourceDescription {
        ResourceDescription::Cidr(ResourceDescriptionCidr {
            id: dns_id(),
            address: addr.parse().unwrap(),
            name: "bar".to_string(),
        })
    }

    fn dns_bar_resource(addr: &str) -> ResourceDescription {
        ResourceDescription::Dns(ResourceDescriptionDns {
            id: dns_id(),
            address: addr.to_string(),
            name: "bar".to_string(),
        })
    }

    fn cidr_baz_resource(addr: &str) -> ResourceDescription {
        ResourceDescription::Cidr(ResourceDescriptionCidr {
            id: additional_id(),
            address: addr.parse().unwrap(),
            name: "baz".to_string(),
        })
    }
```

Might be easier to understand what is going on here if we name the functions uniquely if they have the same ID. Can also go with numbers if you don't want to expose the name but the "additional" is somewhat confusing.